src/Conversion/ONNXToTOSA/Tensor/Reshape.cpp

```diff
@@ -0,0 +1,67 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+//===---------------- Elementwise.cpp - Elementwise Op --------------------===//
+//
+// Copyright (c) 2022 Advanced Micro Devices, Inc.
+//
+// =============================================================================
+//
+// This file lowers ONNX ReshapeOp to TOSA dialect.
+//
+//===----------------------------------------------------------------------===//
+
+#include "mlir/Dialect/Tosa/IR/TosaOps.h"
+#include "mlir/IR/BuiltinAttributes.h"
+#include "mlir/IR/BuiltinTypes.h"
+#include "mlir/IR/TypeUtilities.h"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSACommon.hpp"
+#include "src/Conversion/ONNXToTOSA/ONNXToTOSALegalizeUtils.hpp"
+#include "src/Dialect/ONNX/ONNXOps.hpp"
+#include "llvm/ADT/SmallVector.h"
+
+using namespace mlir;
+
+namespace onnx_mlir {
+
+namespace {
+
+class ONNXReshapeLoweringToTOSA : public OpConversionPattern<ONNXReshapeOp> {
+public:
+  using OpConversionPattern::OpConversionPattern;
+  using OpAdaptor = typename ONNXReshapeOp::Adaptor;
+  LogicalResult matchAndRewrite(ONNXReshapeOp op, OpAdaptor adaptor,
+      ConversionPatternRewriter &rewriter) const override {
+
+    RankedTensorType outputType =
+        op.getResult().getType().dyn_cast<RankedTensorType>();
+
+    if (!outputType) {
+      return rewriter.notifyMatchFailure(op, "Not a ranked tensor");
+    }
+
+    llvm::SmallVector<int64_t> shapeValues;
+    for (int i = 0; i < outputType.getShape().size(); i++) {
+      shapeValues.push_back(outputType.getShape()[i]);
+    }
+    ArrayAttr shapeAttr = rewriter.getI64ArrayAttr(shapeValues);
+
+    auto inputOp = adaptor.shape().getDefiningOp();
+    rewriter.eraseOp(inputOp);
```

Comment:
Why do you need to erase the operator manually?

Reply:
`tosa::CreateReplaceOpAndInfer<tosa::ReshapeOp>` only replaces the reshape op from ONNX. The input constant remains. DCE could fix this, too, but erasing the op seemed to be the right way. @ljfitz what do you think?

Reply:
Ah right, I misread the next, for some reasons I had in mind that it was a CreateOpAndInfer. All good.

Reply:
I deleted the eraseOp. The sce pass later should remove these constants